### PR TITLE
Add shell array parsing helper

### PR DESCRIPTION
**Pull request by schlessera**
Adds two helper functions:

* `WP_CLI\Utils\parse_shell_arrays( $assoc_args, $array_arguments )`
    This takes an existing `$assoc_args` array, and parses possible shell arguments for all keys that are provided in `$array_arguments`

* `WP_CLI\Utils\is_json( $argument, $ignore_scalars = true )`
    Helper function for the above parsing, that detects whether a given string is valid JSON.

See #4616

### CODE REVIEW

File: php/utils.php [hunk 1]
```diff
@@ -1374,3 +1374,49 @@ function _proc_open_compat_win_env( $cmd, &$env ) {
...
+ * Check whether a given string is a valid JSON representation.
+ *
+ * @param string $argument       String to evaluate.
+ * @param bool   $ignore_scalars Optional. Whether to ignore scalar values.
+ *                               Defaults to true.
+ *
+ * @return bool Whether the provided string is a valid JSON representation.
+ */
+function is_json( $argument, $ignore_scalars = true ) {
+	if ( empty( $argument ) || ! is_string( $argument ) ) {
```
Comment (gitlost) [inline php/utils.php]: To be picky this should be `if ( ! is_string( $argument ) || 0 === strlen( $argument ) ) {` to deal with if given `'0'`.

Comment (schlessera) [reply php/utils.php]: Picky is good. Why did you merge anyway?

I'll add a bug report as a good first issue.

Comment (schlessera) [reply php/utils.php]: See https://github.com/wp-cli/wp-cli/issues/4629

Comment (gitlost) [reply php/utils.php]: Ha, I merged because we've so many bugs, PRs and other stuff outstanding and release is 8 days from now!

**Comment by gitlost** [comment]
Removing this from the 1.5.0 milestone in favor of https://github.com/wp-cli/wp-cli/issues/4638

**Pull request merged**